.circleci/config.yml

```
@@ -208,9 +208,44 @@ commands:
 
   run_build:
     steps:
+      # TODO: remove the line below
+      # Before merging this PR we need to use an alternative develop branch to test the functionality, `develop-cache-test`
+
+      # On `develop-cache-test` we intentionally do NOT restore any ccache, so every run starts from scratch.
+      # Other branches restore the most recent cache produced on `develop-cache-test`.
+      - when:
+          condition:
+            not:
+              equal: ["develop-cache-test", << pipeline.git.branch >>]
+          steps:
+            - restore_cache:
+                keys:
+                  # Reuse cache produced on develop-cache-test (prefix match; restores the most recent).
+                  - v1-ccache-{{ arch }}-develop-cache-test-
+                  - v1-ccache-{{ arch }}-
       - run:
```

we usually add a warning here because not everyone may realize that changing the code between `restore_cache` and `save_cache` doe not invalidate the cache automatically.
```suggestion
                  - v1-ccache-{{ arch }}-
      # WARNING! If you edit anything between here and save_cache, remember to bump version or invalidate the cache manually.
      - run:
```